From 66494e6cb0c87aacc9ca80b0cab388ae41929967 Mon Sep 17 00:00:00 2001
From: Scott D Phillips <scott.d.phillips@intel.com>
Date: Mon, 6 Feb 2017 18:35:39 +0000
Subject: [PATCH 2/2] HACK: dmabuf: immediatey release derived image

Original patch received from:
Scott D Phillips <scott.d.phillips@intel.com>

Please note that this hack must be removed once proper
support for dma derived buffers is upstreamed.

Signed-off-by: Soren Friis <soren.friis@intel.com>
---
 gst-libs/gst/vaapi/gstvaapisurface_drm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/vaapi/gstvaapisurface_drm.c b/gst-libs/gst/vaapi/gstvaapisurface_drm.c
index 8c0f8f9..3f0d1bb 100644
--- a/gst-libs/gst/vaapi/gstvaapisurface_drm.c
+++ b/gst-libs/gst/vaapi/gstvaapisurface_drm.c
@@ -38,9 +38,10 @@ gst_vaapi_surface_get_drm_buf_handle (GstVaapiSurface * surface, guint type)
 
   proxy =
       gst_vaapi_buffer_proxy_new_from_object (GST_VAAPI_OBJECT (surface),
-      image->internal_image.buf, type, gst_vaapi_object_unref, image);
+      image->internal_image.buf, type, NULL, NULL);
   if (!proxy)
     goto error_alloc_export_buffer;
+  gst_vaapi_object_unref (image);
   return proxy;
 
   /* ERRORS */
-- 
2.7.4

